src/qibolab/result.py

```diff
@@ -1,53 +1,64 @@
 from dataclasses import dataclass
 from functools import cached_property
+from typing import Optional
 
 import numpy as np
 import numpy.typing as npt
-
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64), ("shots", np.uint32)])
+from scipy import signal
 
 
 @dataclass
 class ExecutionResults:
     """Data structure to deal with the output of execute_pulse_sequence"""
 
-    array: npt.NDArray[ExecRes]
+    _i: npt.NDArray[np.float64]
+    _q: npt.NDArray[np.float64]
+    shots: Optional[npt.NDArray[np.uint32]] = None
 
     @classmethod
-    def from_components(cls, is_, qs, shots):
-        ar = np.empty(len(is_), dtype=ExecRes)
-        ar["i"] = is_
-        ar["q"] = qs
-        ar["shots"] = shots
-        ar = np.rec.array(ar)
-        return cls(ar)
+    def from_components(cls, is_, qs_, shots=None):
+        return cls(is_, qs_, shots)
 
     @property
     def i(self):
-        return self.array.i
+        return self._i
+
+    @i.setter
+    def i(self, value):
+        self._i = value
 
     @property
     def q(self):
-        return self.array.q
+        return self._q
 
-    @property
-    def shots(self):
-        return self.array.shots
+    @q.setter
+    def q(self, value):
+        self._q = value
+
+    def __add__(self, data):
+        i = np.append(self.i, data.i)
+        q = np.append(self.q, data.q)
+
+        new_execution_results = self.__class__.from_components(i, q)
+
+        return new_execution_results
 
     @cached_property
     def msr(self):
         """Computes msr value."""
-        return np.sqrt(self.array.i**2 + self.array.q**2)
+        return np.sqrt(self.i**2 + self.q**2)
 
     @cached_property
     def phase(self):
         """Computes phase value."""
-        return np.angle(self.array.i + 1.0j * self.array.q)
+        phase = np.angle(self.i + 1.0j * self.q)
+        return phase
+        # return signal.detrend(np.unwrap(phase))
```

Comment:
I forgot to write a comment about this in the code. For qblox it doesn't break but the plots are quite weird.
Results with standard phase:
![without_scipy](https://user-images.githubusercontent.com/49183315/217192947-25b52dc5-ae67-4628-a4e9-3e8ca10628e0.png)

Results using `signal.detrend`

![with_scipy](https://user-images.githubusercontent.com/49183315/217193075-a04e1493-94e7-43ed-96d8-09425c6f48c6.png)

For the `fast_sweep` everything is fine, however when we include the precision sweep the results are not consistent with the previous sweep. @stavros11 were you using both the fast and the precision sweep with QM?

I have no problems in keeping this also for qblox. Moreover, if in the near future we will drop one of the two sweeps the results are perfectly acceptable also for qblox.

Let me know what you think.

Reply:
> I forgot to write a comment about this in the code.

Docstrings are better than comments

Reply:
> For the `fast_sweep` everything is fine, however when we include the precision sweep the results are not consistent with the previous sweep. @stavros11 were you using both the fast and the precision sweep with QM?

Yes, I am using both sweeps. For QM the data looks like that:

1. Without any processing ![Screenshot from 2023-02-10 13-18-13](https://user-images.githubusercontent.com/35475381/218053281-1dba0f03-068d-48a3-9493-e89e4b262dfb.png)
2. With `np.unwrap` ![Screenshot from 2023-02-10 13-18-26](https://user-images.githubusercontent.com/35475381/218053384-85a6277f-1912-4352-9b1d-e484875784d7.png)
3. With `signal.detrend` ![Screenshot from 2023-02-10 13-18-33](https://user-images.githubusercontent.com/35475381/218053477-5c0bc953-e59e-43f0-a6f4-445b8c5b78b1.png)

however I only see this in the resonator spectroscopy. For qubit spectroscopy and other calibrations I do not see any qualitative difference in the plot shape:

<details>
<summary>Qubit spectroscopy plots</summary>

![Screenshot from 2023-02-10 13-23-31](https://user-images.githubusercontent.com/35475381/218053884-5757eea8-7fec-4a40-b1e4-ec0711cf0b16.png)

![Screenshot from 2023-02-10 13-23-42](https://user-images.githubusercontent.com/35475381/218053924-59e6760e-eebf-4b40-bec5-f88c1b5217cc.png)

</details>

it is only the phase numbers (y-axis) that differ.

There should be a simple explanation for this, I am just not very familiar with the theory of i-q signals. I guess it appears mostly on the resonator spectroscopy because we are sweeping the readout frequency which affects the readout phase (phase ~ w t). In practice, since the phase is not super-relevant for what we are doing, either way it would work. Maybe not doing any transformation is better, as we can always do it later in the plot if really needed.

Reply:
Thanks for the feedback @stavros11.
At this point I would suggest to not do any transformations. If a particular user would like to apply some post-processing operation such as `np.unwrap` or `signal.detrend` he can always use the raw data in order to generate new plots.

Reply:
Let me know if you agree.

Reply:
You may only want to remove the scipy import.